contents/scalability/scalable-development.md

```diff
@@ -33,9 +33,40 @@ There are a few common bottlenecks that can be avoided such as database querying
+1. More details on the common performance bottlenecks ([resource](https://www.apicasystem.com/blog/5-common-performance-bottlenecks))
 
-Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases/)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+1. Relational vs Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+
+### Caching
+With an increasing user base, your server has to deal with a larger number of requests along with the bottlenecks mentioned previously such as network congestion and database querying. As a result, the response becomes slower. This is where caching comes in order to meet this demand. A cache is a key-value store that resides between the application and the database which can either be in the browser or part of your server infrastructure itself. By retrieving data from the cache instead of the database, the response time reduces greatly.
+
+The next thing to consider is what to cache. The rule of thumb is to cache data that is frequently accessed and read more often than it is updated. With this, cache hits would be more often than cache misses and as a result, the time saved from faster accesses and reads outweighs the extra time taken to populate the cache.  
+
+1. Scalability using caches ([resource](http://www.lecloud.net/post/9246290032/scalability-for-dummies-part-3-cache))
+
+1. Scalability best practices ([resource](https://www.infoq.com/articles/ebay-scalability-best-practices))
+
+### Sharding 
+As your traffic increases your data increases as well and as a result, your database gets overloaded. One way to mitigate this is to scale your database by sharding. Sharding is a method of splitting and storing a single logical dataset in multiple databases. More specifically, it is the storing of data horizontally - storing rows of a same table in multiple database nodes instead of storing them in the usual vertical way - storing different tables & columns in a separate database.
+
+Sharding is essential when your dataset becomes too large to store in a single database. It reduces the number of rows in each table and as such improves search performance since the search is done on a smaller table. 
+
+Querying the databases containing only the relevant partitions becomes possible as well. For instance, if a database contains a column for age, then you can partition the rows according to an age group and store them in different databases. Whenever there is a need to access the data of a particular age group, instead of querying the whole database, you just need to query the partition that contains that age group. This allows your database to scale along with your data and traffic growth.
```

Comment:
If there are two benefits, it's clearer if you gave them as a numbered list or a bullet list rather than two paragraphs.

Reply:
The term `partition` is being used without explanation?

Reply:
I think the two points here are a bit tangled with each other; you can try to tease them apart.
1. sharding can store more data than a single database can hold
2. sharding can make queries faster

Reply:
>The term partition is being used without explanation?

In this case, partition just means to divide into parts. Since it's not really a technical term, there's no need for an explanation right?